File: src/main/resources/example/scala/ExamplePlugin.scala
import javax.inject.Inject

import org.slf4j.Logger
import org.spongepowered.api._
import org.spongepowered.api.event._
import org.spongepowered.api.event.state._
import org.spongepowered.api.plugin._
import org.spongepowered.api.service.config.DefaultConfig

@Plugin()
class ExamplePlugin {

  @Inject private val game: Game
  @Inject private val logger: Logger

  @Inject
  @DefaultConfig(sharedRoot = true)
  private val configLoader: ConfigurationLoader[CommentedConfigurationNode]

  @Subscribe
  def onStarting(event: ServerStartingEvent) {
  }

  @Subscribe
  def onPreInit(event: PreInitializationEvent) {
  }

  @Subscribe
  def onInit(event: InitializationEvent) {
  }

  @Subscribe
  def disable(event: ServerStoppingEvent) {
  }

}

File: src/main/resources/example/java/ExamplePlugin.java
@Plugin(id = "@[plugin.id]", name = "@[plugin.name]", version = "@[plugin.version]")
public class ExamplePlugin {

    @Inject private Game game;
    @Inject private Logger logger;
    @Inject @DefaultConfig(sharedRoot = true) private ConfigurationLoader<CommentedConfigurationNode> configLoader;

    @Subscribe
    public void onStarting(ServerStartingEvent event) {
    }

    @Subscribe
    public void onPreInit(PreInitializationEvent event) {
    }

    @Subscribe
    public void onInit(InitializationEvent event) {
    }

    @Subscribe
    public void disable(ServerStoppingEvent event) {
    }

}